src/NATS.Client.ObjectStore/Internal/Encoder.cs

```diff
@@ -68,18 +69,33 @@ public static string Encode(string arg)
     /// <exception cref="ArgumentNullException">'inArray' is null.</exception>
     /// <exception cref="ArgumentOutOfRangeException">offset or length is negative OR offset plus length is greater than the length of inArray.</exception>
     public static string Encode(Span<byte> inArray, bool raw = false)
+    {
+        using (var owner = EncodeToMemoryOwner(inArray, raw))
+        {
+            var segment = owner.DangerousGetArray();
+            if (segment.Array == null || segment.Array.Length == 0)
+            {
+                return string.Empty;
+            }
+
+            return new string(segment.Array, segment.Offset, segment.Count);
+        }
+    }
+
+    public static NatsMemoryOwner<char> EncodeToMemoryOwner(Span<byte> inArray, bool raw = false)
     {
         var offset = 0;
         var length = inArray.Length;
 
         if (length == 0)
-            return string.Empty;
+            return NatsMemoryOwner<char>.Empty;
 
         var lengthMod3 = length % 3;
         var limit = length - lengthMod3;
-        var output = new char[(length + 2) / 3 * 4];
+        var owner = NatsMemoryOwner<char>.Allocate((length + 2) / 3 * 4);
```

Comment:
> Should we be try-catch-dispose-throw around this, case something derps in the encode?

Should we encode to span instead? in case of sha we can also stackalloc maybe?

> Also, Should we be adding a flag to the method here to specify 'clearing' and pass it in here?

Do you mean clearing the input array or output?

Reply:
> Should we encode to span instead? in case of sha we can also stackalloc maybe?

Hmm I can give it a shot, it might be cleaner...

> Do you mean clearing the input array or output?

Clearing the buffered array. IIRC NatsMemoryOwner has a flag to say whether the array is cleared on return.

My 'gut' says yes but if we can get away with stackallocing the span, point is moot. Will check.